packages/truffle-db/src/workspace/test/index.ts

```diff
@@ -14,27 +18,74 @@ const fixturesDirectory = path.join(
   "fixtures"
 );
 
+class WorkspaceClient {
+  private workspace: Workspace;
+
+  constructor () {
+    this.workspace = new Workspace();
+  }
+
+  async execute (request, variables = {}) {
+    const result = await graphql.execute(
+      schema,
+      request,
+      null, // root object, managed by workspace
+      { workspace: this.workspace }, // context vars
+      variables
+    );
+
+    return result.data;
+  }
+}
+
 const Migrations = require(path.join(fixturesDirectory, "Migrations.json"));
 
-const GetContractNames = `
+let sourceId;
+let bytecodeId;
+let compilationId;
+let contractId;
+
+beforeAll(() => {
+  sourceId = soliditySha3(Migrations.source, Migrations.sourcePath);
+  bytecodeId = soliditySha3(Migrations.bytecode);
+  compilationId = soliditySha3(jsonStableStringify({ 
+    compiler: Migrations.compiler, 
+    sourceIds: [{ id: sourceId }] 
+  }));
+  contractId = soliditySha3(jsonStableStringify({ 
+    name: Migrations.contractName, 
+    abi: { json: JSON.stringify(Migrations.abi) } , 
+    sourceContract: { index: 0 } ,
+    compilation: { id: compilationId }
+  }));
+
+});
```

Comment:
Maybe instead of a `beforeAll` block, we could just make a helper function, something like:

```javascript
const generateId = (obj) => soliditySha3(jsonStableStringify(obj))
```

I mention this because I prefer not to generate the IDs outside the tests. I'll expand on this below per-test.